src/main/scala/intellij/haskell/action/DownloadCabalConfigFile.scala

```diff
@@ -0,0 +1,17 @@
+package intellij.haskell.action
+
+import com.intellij.openapi.actionSystem.{AnAction, AnActionEvent}
+import intellij.haskell.external.component.HaskellComponentsManager.refreshGlobalProjectInfo
+import intellij.haskell.external.component.PathComponent
+
```

Comment:
Ok, so maybe download file [here](https://github.com/zjhmale/intellij-haskell/blob/cabal-completion/src/main/scala/intellij/haskell/external/component/GlobalProjectInfoComponent.scala#L58) in GlobalProjectInfo is cool enough? Cause the global project info will always be loaded when a project is created or opened right?
>I think best solution is to download file when project is created or opened.

I think the reason why we need this action is when the network is blocked maybe users can not download file at first, so when the network is available, users can use this action to download the config file manually and refresh the cache to make this completion working right.
>I do not see reason to create action for downloading cabal.config file. It should be downloaded when it is necessary.